src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -878,7 +878,7 @@ void ShenandoahHeap::notify_heap_changed() {
   // Update monitoring counters when we took a new region. This amortizes the
   // update costs on slow path.
   monitoring_support()->notify_heap_changed();
-  _heap_changed.set();
+  _heap_changed.try_set();
```

Comment:
I'm not sure I understand the significance of this change from .set() to .try_set().

Reply:
Would a comment help explain why it is ok for set() to fail here?

Reply:
It was an unnecessary change. Upstream has `try_set`. We just need one thread to set it. `try_set` avoid modifying the shared variable if it is already set.